managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/MetaStoreImplZookeeper.java

```diff
@@ -297,5 +304,63 @@ public void removeManagedLedger(String ledgerName, MetaStoreCallback<Void> callb
         }
     }
 
+    private ManagedLedgerInfo parseManagedLedgerInfo(byte[] data)
+            throws ParseException, InvalidProtocolBufferException {
+        if (protobufFormat == ZNodeProtobufFormat.Text) {
+            // First try text format, then fallback to binary
+            try {
+                return parseManagedLedgerInfoFromText(data);
+            } catch (ParseException e) {
+                return parseManagedLedgerInfoFromBinary(data);
+            }
+        } else {
+            // First try binary format, then fallback to text
+            try {
+                return parseManagedLedgerInfoFromBinary(data);
+            } catch (InvalidProtocolBufferException e) {
+                return parseManagedLedgerInfoFromText(data);
+            }
+        }
+    }
+
+    private ManagedLedgerInfo parseManagedLedgerInfoFromText(byte[] data) throws ParseException {
```

Comment:
do you think is it fine to combine two ledger/cursor parse method to:
`parseInfoFromText(final byte[] data, final Message.Builder builder)`